src/passes/OptimizeInstructions.cpp

```diff
@@ -3549,6 +3558,52 @@ struct OptimizeInstructions
     return nullptr;
   }
 
+  // Bitwise AND of a value with bits in [0, n) and a constant with no bits in
+  // [0, n) always yields 0. Replace with zero.
+  Expression* optimizeAndNoOverlappingBits(Binary* curr) {
+    assert(curr->op == AndInt32 || curr->op == AndInt64);
+
+    using namespace Abstract;
+    using namespace Match;
+
+    auto* left = curr->left;
+    auto* right = curr->right;
+
+    // Check right as constant and left's max bits
```

Comment:
The comment here is in the opposite order of the code. I would reverse it, to follow the code.